Dockerfile

```diff
@@ -11,7 +11,9 @@ FROM alpine:3.9
 # experience when this doesn't work out of the box.
 #
 # OpenSSL is required so wget can query HTTPS endpoints for health checking.
-RUN apk add --update ca-certificates openssl
+RUN apk add --update ca-certificates openssl\
+  && addgroup --gid 1001 dex\
+  && adduser --disabled-password --no-create-home --home "$(pwd)" -u 1001 -G dex dex
```

Comment:
nit: dex doesn't need a home directory, why provide `--home "$(pwd)"`?

Reply:
It will set home dir to `/`. Even users with `nologin` have some home directories:
```
daemon:x:1:1:daemon:/usr/sbin:/usr/sbin/nologin
bin:x:2:2:bin:/bin:/usr/sbin/nologin
sys:x:3:3:sys:/dev:/usr/sbin/nologin
```
AFAIK if the `--home` argument is not provided it will set home to `/home/dex` which will be non-existing directory. Setting home to `$(pwd)` ensures that the directory exists.